metadata-ingestion/src/datahub/utilities/file_backed_collections.py

```diff
@@ -255,16 +257,44 @@ def __delitem__(self, key: str) -> None:
             raise KeyError(key)
 
     def __iter__(self) -> Iterator[str]:
+        # Cache should be small, so safe list cast to avoid mutation during iteration
+        cache_keys = list(self._active_object_cache.keys())
+        yield from cache_keys
+
         cursor = self._conn.execute(f"SELECT key FROM {self.tablename}")
         for row in cursor:
-            if row[0] in self._active_object_cache:
-                # If the key is in the active object cache, then SQL isn't the source of truth.
-                continue
+            if row[0] not in cache_keys:
+                yield row[0]
+
+    def items(self) -> Iterator[Tuple[str, _VT]]:
+        # Implement manually to avoid __getitem__ calls
+        cache_keys = list(self._active_object_cache.keys())
+        yield from self._active_object_cache.items()
+
+        cursor = self._conn.execute(f"SELECT key, value FROM {self.tablename}")
+        for row in cursor:
+            if row[0] not in cache_keys:
+                yield row[0], self.deserializer(row[1])
```

Comment:
this means that we're not tracking these objects in the active cache, which means that mutations made against items() won't get saved